srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.xml

```diff
@@ -384,7 +384,17 @@ EKTField = FullEKTField / ShortEKTField / ExtensionEKTField
               ShortEKTField, the ShortEKTField is removed from the
               packet then normal SRTP or SRTCP processing occurs. If
               the packet contains a FullEKTField, then processing
-              continues as described below.</t>
+              continues as described below.
+              The reason for using the last byte of the packet to
+              indicate the type is that the length of the SRTP or
+              SRTCP part is not known until the decryption has
+              occurred. At this point in the processing, there is no
+              easy way to know where the EKT field would
+              start. However, the whole UDP packet has been received
+              so instead of the starting at the front of the packet,
+              the parsing work backwards off the end of the packet and
+              thus the type is put at the very end of the packet.
```

Comment:
The type doesn't indicate length, though.  The length does.

The type just tells us whether it's a short tag or long tag (or some TBD tag).  The length precedes the type in the long tag.

So, the type is there to tell us if we have a short tag (one octet) or another tag type.  In the latter case, an EKT tag length is introduced  just before the type field.